manifests/params.pp

```diff
@@ -3,7 +3,39 @@
 
   case $::osfamily {
     'RedHat': {
-      if versioncmp($::operatingsystemmajrelease, '7') == 0 {
+      if versioncmp($::operatingsystemrelease, '7.4.1708') >= 0 {
```

Comment:
The operatingsystemrelease in RHEL is just 7.4.  So this condition should probably just be checking against 7.4.

Reply:
Have you verified on actual RHEL-7.4, because I get this on CentOS-7.4 (and do not have a RHEL-7.4 at my disposal currently):

```
[root@anode001 vagrant]# puppet --version
5.2.0
[root@anode001 vagrant]# facter --version
3.9.0 (commit 1650f446b4b6076ab0e5937ef9e47a4523fc385b)

[root@anode001 vagrant]# facter operatingsystemrelease
7.4.1708
[root@anode001 vagrant]# facter os
{
  architecture => "x86_64",
  family => "RedHat",
  hardware => "x86_64",
  name => "CentOS",
  release => {
    full => "7.4.1708",
    major => "7",
    minor => "4"
  },
  selinux => {
    enabled => false
  }
}
``

I'll verify the "probably" tomorrow when I'm back at work. Nice evening.

Reply:
Ah, I see. You are referring to the failing tests. As it seems, a correct catalog gets applied correctly using the correct packages but the spec tests are still comparing the state against the package lists of 7.0, which is wrong.

Reply:
@mrolli RHEL 7.4 shows as just 7.4.

```
# cat /etc/redhat-release 
Red Hat Enterprise Linux Server release 7.4 (Maipo)
# /opt/puppetlabs/bin/facter -p os.release
{
  full => "7.4",
  major => "7",
  minor => "4"
}
# /opt/puppetlabs/bin/facter -p operatingsystemrelease
7.4

```

Reply:
Interesting difference between the two. Thanks for the output. I changed the release number to 7.4. Works for both then.